packages/react-reconciler/src/ReactFiberCompleteWork.old.js

```diff
@@ -679,7 +679,7 @@ function completeWork(
           // If we hydrated, then we'll need to schedule an update for
           // the commit side-effects on the root.
           markUpdate(workInProgress);
-        } else if (!fiberRoot.hydrate) {
+        } else if (!fiberRoot.hydrate && workInProgress.child !== null) {
```

Comment:
Hm...so the way this check is supposed to work is this:

If the current Fiber didn't render any children (`current.child === null`) then that means we are either mounting (in which case the container should be clear, or we should clear it) or updating (in which case, if `current.child` is null, our previous render didn't render any children- so clearing the container should be a noop anyway).

The work in progress fiber having a child just means "we are about to add children to this container".

Reply:
Shouldn't the work in progress have children? If it doesn't, aren't we rendering nothing over previously having nothing?

Reply:
Whether the work in progress has children is not relevant to the specific question of whether the container has non-React contents in it. That's what we're trying to clear here. If we know that React didn't add things to this container the last time we committed (or if we are mounting) then we clear the container to make sure there's nothing unexpected in it.